vm/src/obj/objstr.rs

```diff
@@ -407,8 +407,16 @@ impl PyString {
 
     #[pymethod]
     fn capitalize(&self) -> String {
-        let (first_part, lower_str) = self.value.split_at(1);
-        format!("{}{}", first_part.to_uppercase(), lower_str)
+        let mut char_indices = self.value.char_indices();
+        if let Some((_, first_char)) = char_indices.next() {
+            format!(
+                "{}{}",
+                first_char.to_uppercase(),
+                &self.value[first_char.len_utf8()..].to_lowercase(),
+            )
```

Comment:
You could just use `chars()` for this:
```suggestion
        let mut chars = self.value.chars();
        if let Some(first_char) = chars.next() {
            format!(
                "{}{}",
                first_char.to_uppercase(),
                &chars.as_str().to_lowercase(),
            )
```